example/federation/products/graph/generated/federation.go

```diff
@@ -75,36 +76,52 @@ func (ec *executionContext) __resolve_entities(ctx context.Context, representati
 		}()
 
 		switch typeName {
-
 		case "Manufacturer":
-			id0, err := ec.unmarshalNString2string(ctx, rep["id"])
-			if err != nil {
-				return errors.New(fmt.Sprintf("Field %s undefined in schema.", "id"))
-			}
+			entity, err := func() (*model.Manufacturer, error) {
+				id0, err := ec.unmarshalNString2string(ctx, rep["id"])
+				if err == nil {
+					return ec.resolvers.Entity().FindManufacturerByID(ctx, id0)
+				}
+				return nil, nil
+			}()
 
-			entity, err := ec.resolvers.Entity().FindManufacturerByID(ctx,
-				id0)
 			if err != nil {
-				return err
+				return fmt.Errorf(`resolving Entity "Manufacturer": %w`, err)
+			}
+			if entity == nil {
+				return errors.New(`unable to resolve Entity "Manufacturer"`)
 			}
 
 			list[idx[i]] = entity
 			return nil
 
 		case "Product":
-			id0, err := ec.unmarshalNString2string(ctx, rep["manufacturer"].(map[string]interface{})["id"])
-			if err != nil {
-				return errors.New(fmt.Sprintf("Field %s undefined in schema.", "manufacturerID"))
-			}
-			id1, err := ec.unmarshalNString2string(ctx, rep["id"])
-			if err != nil {
-				return errors.New(fmt.Sprintf("Field %s undefined in schema.", "id"))
+			entity, err := func() (*model.Product, error) {
+				id0, err := ec.unmarshalNString2string(ctx, rep["manufacturer"].(map[string]interface{})["id"])
```

Comment:
This is the other issue.  If the query entity was for `FindProductByUpc` and there is no `manufacturer` representation, than this will fail because we would be trying to read `id` from `nil`.

https://go.dev/play/p/yLWsp2g0gyR